rs/p2p/consensus_manager/src/receiver.rs

```diff
@@ -364,14 +365,25 @@ where
                     (false, None)
                 }
             }
-            Entry::Vacant(empty_slot) => {
+            // Only insert slot update if we are below peer slot table limit.
+            Entry::Vacant(empty_slot) if peer_slot_table_len < self.slot_limit => {
                 empty_slot.insert(new_slot_entry);
                 self.metrics
                     .slot_table_new_entry_total
                     .with_label_values(&[peer_id.to_string().as_str()])
                     .inc();
                 (true, None)
             }
+            Entry::Vacant(_) => {
+                self.metrics.slot_table_limit_exceeded_total.inc();
+                warn!(
+                    self.log,
+                    "Peer {} tries to exceed slot limit {}. Dropping slot update",
+                    peer_id,
+                    self.slot_limit
+                );
+                return;
```

Comment:
tiny nit: to be consistent with a branch above (line 365)
```suggestion
                (false, None)
```